e4s_cl/cf/wi4mpi/install.py

```diff
@@ -0,0 +1,168 @@
+    if vendor_list:
+        vendor = vendor_list[0][0]
+        installed = DISTRO_DICT.get(vendor)()
+    return installed
+
+
+def download_wi4mpi(url: str, destination: Path) -> Optional[Path]:
+    """
+    Download and extract the TAR archive from 'url' into the directory 'destination'
+    """
+    with urllib.request.urlopen(url) as request:
+        if not request.status == 200:
+            LOGGER.error("Failed to download Wi4MPI release; aborting")
+            return None
+
+        with NamedTemporaryFile(delete=False) as buffer:
+            buffer.write(request.read())
+            archive = buffer.name
+
+    if not tarfile.is_tarfile(archive):
+        LOGGER.error("Downloaded file is not an archive; aborting")
+        return None
+
+    with tarfile.open(archive) as data:
+        if not safe_tar(data):
+            LOGGER.error("Unsafe paths detected in archive; aborting")
+            return None
+
+        data.extractall(destination)
+
+        release_root_dir = min(data.getnames())
+
+    try:
+        Path(archive).unlink()
+    except OSError as err:
+        LOGGER.error("Failed to delete downloaded data: %s", str(err))
+
+    return destination / release_root_dir
+
+
+def install_wi4mpi() -> bool:
+    """Clones and installs wi4mpi from git run
+    
+    Installs in ~/.local/share/wi4mpi using a GNU compiler
+    """
+
+    cmake_executable = which("cmake")
+    if not cmake_executable:
+        LOGGER.warning(
+            "WI4MPI installation failed: cmake is missing. Proceeding with profile initialisation"
+        )
+        return False
+
+    def _run_wi4mpi_install_cmd(cmd, discard_output=True):
+        """
+        Run a command. Disables output on discard_output=True
+        """
+        # None is the default; no redirection
+        stdout, stderr = None, None
+        if discard_output:
+            # Redirect output to null and error to output (to null)
+            stdout, stderr = DEVNULL, STDOUT
+
+        with subprocess.Popen(cmd, stdout=stdout, stderr=stderr) as proc:
+            status = proc.wait()
+        return status == 0
+
+    def _double_tap(cmd):
+        """
+        Run a given command (cmake/make) discarding the output. If the
+        returncode indicates an error, run it again with the out/err streams
+        enabled; this ensures a concise error output as recommended on the GNU
+        make's website.
+        """
+
+        success = _run_wi4mpi_install_cmd(cmd, discard_output=True)
+        if not success:
+            _run_wi4mpi_install_cmd(cmd, discard_output=False)
+
+        return success
+
+    build_dir = WI4MPI_DIR / 'build'
+    source_dir = download_wi4mpi(WI4MPI_RELEASE_URL, WI4MPI_DIR)
+    if source_dir is None:
+        LOGGER.error("Failed to access Wi4MPI release")
+        return False
+
+    configure_cmd = [
+        cmake_executable, \
+        '-DCMAKE_INSTALL_PREFIX=~/.local/wi4mpi', \
+        '-DWI4MPI_COMPILER=GNU', \
```

Comment:
This should be a dynamic line to ensure the flag reflects what is available on the system

Reply:
After testing, this is no longer the desired solution as Wi4MPI requires a full install with `fortran` support, and this is not a given if we compile with the same vendors as the MPI libraries. What to do  ?